server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -276,19 +259,16 @@ public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception
             queryLatch.countDown();
             assertBusy(() -> {
                 final List<SearchTask> searchTasks = getSearchTasks();
-                // The search request can complete before the "cancelledLatch" is latched if the second shard request is sent
-                // after the request was cancelled (i.e., the child task is not allowed to start after the parent was cancelled).
                 if (searchTasks.isEmpty() == false) {
```

Comment:
I believe the current assertion of the test is too lenient (my fault). We should strengthen it. It should ensure that the search task and its child tasks are canceled when the coordinator receives a failure from a target shard.